src/codegen/compile.ml

```diff
@@ -8930,25 +8847,9 @@ module FuncDec = struct
       | Type.Shared Type.Query ->
         Lifecycle.trans env Lifecycle.PostQuery
       | Type.Shared Type.Composite ->
-        (* Stay in composite query state such that callbacks of 
-        composite queries can also use the memory reserve. 
-        The state is isolated since memory changes of queries 
-        are rolled back by the IC runtime system. *)
-        Lifecycle.trans env Lifecycle.InComposite
+        Lifecycle.trans env Lifecycle.Idle
       | _ -> assert false
 
-  let callback_start env =
-    Lifecycle.is_in env Lifecycle.InComposite ^^
-    G.if0
-      (G.nop)
-      (message_start env (Type.Shared Type.Write))
-
-  let callback_cleanup env =
-    Lifecycle.is_in env Lifecycle.InComposite ^^
-    G.if0
-      (G.nop)
-      (message_cleanup env (Type.Shared Type.Write))
-  
   let compile_const_message outer_env outer_ae sort control args mk_body ret_tys at : E.func_with_names =
```

Comment:
Are these Lifestyle changes deliberate or  a bad merge?

Reply:
This is because I removed the memory reserve for queries, upgrades, and composite queries. I thought 64-bit space would make this issue redundant or we should address this on the IC. But maybe I should reintroduce that memory reserve functionality.

Reply:
No that makes sense. If we got for a single compile.ml (more on that later)
, we might need it back again...